### PR TITLE
Add `drop_nil_fields` option

### PR DESCRIPTION
**Pull request by nvieirafelipe**

### CODE REVIEW

File: lib/segment/encoder.ex [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  """
+  @spec encode!(struct(), list()) :: String.t()
+  def encode!(struct, options) do
+    struct
+    |> maybe_drop_nil_fields(options)
+    |> Poison.encode!()
+  end
+
+  defp maybe_drop_nil_fields(struct, options) do
```
Comment (prodis) [inline lib/segment/encoder.ex:39]: Missing typespec. The same for the other private functions.

File: lib/segment/encoder.ex [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    if Keyword.get(options, :drop_nil_fields) == true do
```
Comment (prodis) [inline lib/segment/encoder.ex:40]: Is it necessary to compare with `true`? I think `if Keyword.get(options, :drop_nil_fields) do` would have the same result.

File: test/segment/encoder_test.exs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    test "transforms a struct into a JSON string", %{batch: batch} do
+      expected_response =
+        :batch
+        |> Factory.map_for()
+        |> Poison.encode!()
+
+      assert Subject.encode!(batch, []) == expected_response
+    end
+
+    test "when `drop_nil_fields` options is `true`, " <>
```
Comment (prodis) [inline test/segment/encoder_test.exs:26]: Missing a test scenario where `drop_nil_fields` is present but not `true`.

**Review by prodis — APPROVED** [review]
LGTM

Please see my comments.

File: lib/segment/encoder.ex (hunk 2 above)
Comment (nvieirafelipe) [reply lib/segment/encoder.ex:40]: Not necessary, but if we don't compare it a `Analytics.call(something, drop_nil_fields: "Please don't")` will also enable the feature 😄

File: lib/segment/encoder.ex (hunk 1 above)
Comment (nvieirafelipe) [reply lib/segment/encoder.ex:39]: I'll add it but remember: we don't have dialyzer in this project so there's no way to check if the typespec is correct 🤷

File: lib/segment/encoder.ex [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  @moduledoc """
+  Responsible to transforming structs into JSON strings.
+  """
+
+  @doc """
+  Encodes a given struct into a JSON string.
+
+  ## Options
+
+      * `drop_nil_fields`: If set to `true` all the struct `nil` fileds will be
```
Comment (sergioaugrod) [inline lib/segment/encoder.ex]:
```suggestion
      * `drop_nil_fields`: If set to `true` all the struct `nil` fields will be
```

**Review by sergioaugrod — APPROVED** [review]

**Pull request merged**